kiwipy/rmq/tasks.py

```diff
@@ -63,16 +60,22 @@ def __init__(self,
         self._consumer_tag = None
 
         self._task_queue = None  # type: Optional[aio_pika.Queue]
-        self._subscribers = []
+        self._subscribers = {}
         self._pending_tasks = []
 
-    async def add_task_subscriber(self, subscriber):
-        self._subscribers.append(subscriber)
+    async def add_task_subscriber(self, subscriber, identifier=None):
+        identifier = identifier or shortuuid.uuid()
+        if identifier in self._subscribers:
+            raise kiwipy.DuplicateSubscriberIdentifier("Task identifier '{}'".format(identifier))
+
+        self._subscribers[identifier] = subscriber
         if self._consumer_tag is None:
             self._consumer_tag = await self._task_queue.consume(self._on_task)
 
-    async def remove_task_subscriber(self, subscriber):
-        self._subscribers.remove(subscriber)
+        return identifier
+
+    async def remove_task_subscriber(self, identifier):
+        self._subscribers.pop(identifier)
```

Comment:
you want to catch the possible `KeyError` here or just let it bubble up?

Reply:
Done.